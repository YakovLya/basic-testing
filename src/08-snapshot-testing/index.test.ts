import { generateLinkedList } from './index';

describe('generateLinkedList', () => {
  // Check match by expect(...).toStrictEqual(...)
  test('should generate linked list from values 1', () => {
    const list = generateLinkedList(['first elem', 2]);
    expect(list).toStrictEqual({
      value: 'first elem',
      next: {
        value: 2,
        next: {
          value: null,
          next: null,
        },
      },
    });
  });

  // Check match by comparison with snapshot
  test('should generate linked list from values 2', () => {
    const list = generateLinkedList(['first elem', 2, { value: 'elem 3' }]);
    expect(list).toMatchSnapshot();
  });
});
