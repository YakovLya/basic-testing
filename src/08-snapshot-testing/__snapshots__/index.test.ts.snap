// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateLinkedList should generate linked list from values 2 1`] = `
{
  "next": {
    "next": {
      "next": {
        "next": null,
        "value": null,
      },
      "value": {
        "value": "elem 3",
      },
    },
    "value": 2,
  },
  "value": "first elem",
}
`;
